moto/kinesis/responses.py

```diff
@@ -24,7 +25,9 @@ def is_firehose(self):
     def create_stream(self):
         stream_name = self.parameters.get('StreamName')
         shard_count = self.parameters.get('ShardCount')
-        self.kinesis_backend.create_stream(stream_name, shard_count, self.region)
+        stream = self.kinesis_backend.create_stream(stream_name, shard_count, self.region)
+        if isinstance(stream, BadRequest):
```

Comment:
You are right, it should raise a exception in `self.kinesis_backend.create_stream`.
I'll update the request